### PR TITLE
Error and sheet crash reduction

### PR DESCRIPTION
**Pull request by naterush**
# Description

Makes the changes to fix errors and sheet crash reduction found in this investigation: https://www.notion.so/trymito/3-20-22-86a74cf19dc946379f6f60ac109fde53

I did this process a bit different than described in the initial doc. For sheet crashing, looked at the last week of sheet crashes. Turns out these are much harder to get to the bottom of now, as I've got most of the easy ones, so many of them simply cannot be backtracked (and we need to think of other strats for avoiding these issues).

For errors, the categorization I proposed initially (our fault, theirs, etc) seemed mostly not helpful. With 90% of errors, we know about them, they are from user input, or we're handling them in some way already. The other 10% were clear "needs investigation."

I like this new structure of (sheet crashing => address the ones I can) (errors => investigate the ones that clearly need it). Let's do this going forward - I think over time it will lead to dramatically more robust software!

# Testing

Make sure these changes don't break anything.

# Documentation

No.

### CODE REVIEW

**Review by aarondr77 — COMMENTED** [review]
Nice investigation.

File: mitosheet/src/components/Mito.tsx [hunk 1]
```diff
@@ -222,21 +222,21 @@ export const Mito = (props: MitoProps): JSX.Element => {
         previousNumSheetsRef.current = sheetDataArray.length;
     }, [sheetDataArray])
 
-    const previousNumGraphsRef = useRef<number>(Object.keys(analysisData.graphDataDict).length)
+    const previousNumGraphsRef = useRef<number>(Object.keys(analysisData.graphDataDict || {}).length)
     const previousGraphIndex = useRef<number>(uiState.selectedGraphID !== undefined ?
-        Object.keys(analysisData.graphDataDict).indexOf(uiState.selectedGraphID) : -1)
+        Object.keys(analysisData.graphDataDict  || {}).indexOf(uiState.selectedGraphID) : -1)
```
[the remark this answers is not in the document]
Comment (naterush) [reply mitosheet/src/components/Mito.tsx:227]: Was this addressed in our conversation?

File: mitosheet/mitosheet/step_performers/column_steps/change_column_dtype.py [hunk 2]
```diff
@@ -124,7 +124,7 @@ def execute( # type: ignore
                 if is_bool_dtype(new_dtype):
                     new_column = column.fillna(False).astype('bool')
                 elif is_int_dtype(new_dtype):
-                    new_column = column.astype('int')
+                    new_column = column.fillna(0).astype('int')
```
[the remark this answers is not in the document]
Comment (naterush) [reply mitosheet/mitosheet/step_performers/column_steps/change_column_dtype.py:127]: @aarondr77 we can remove this one if you want. I feel like it's probably fine, personally, but your call. Let me know and I'll remove it.

We could also just add a subtext in the case that it's a float column that says "Casting to an int will turn all NaN values to 0." Thoughts?

[the remark this answers is not in the document]
Comment (aarondr77) [reply mitosheet/mitosheet/step_performers/column_steps/change_column_dtype.py:127]: That sounds good to me

File: mitosheet/src/components/Mito.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (aarondr77) [reply mitosheet/src/components/Mito.tsx:227]: Yeah!

**Pull request merged**